cmd/gcp-routes-controller/run.go

```diff
@@ -0,0 +1,181 @@
+
+	health "github.com/InVisionApp/go-health"
+	"github.com/InVisionApp/go-health/checkers"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/machine-config-operator/pkg/version"
+)
+
+var (
+	runCmd = &cobra.Command{
+		Use:   "run",
+		Short: "Runs the gcp-routes-controller",
+		Long:  "",
+		RunE:  runRunCmd,
+	}
+
+	runOpts struct {
+		gcpRoutesService string
+		rootMount        string
+
+		healthCheckURL string
+	}
+)
+
+func init() {
+	rootCmd.AddCommand(runCmd)
+	runCmd.PersistentFlags().StringVar(&runOpts.gcpRoutesService, "gcp-routes-service", "gcp-routes.service", "The name for the service controlling gcp routes on host")
+	runCmd.PersistentFlags().StringVar(&runOpts.rootMount, "root-mount", "/rootfs", "where the nodes root filesystem is mounted for chroot and file manipulation.")
+	runCmd.PersistentFlags().StringVar(&runOpts.healthCheckURL, "health-check-url", "", "HTTP(s) URL for the health check")
+}
+
+func runRunCmd(cmd *cobra.Command, args []string) error {
+	flag.Set("logtostderr", "true")
+	flag.Parse()
+
+	// To help debugging, immediately log version
+	glog.Infof("Version: %+v (%s)", version.Raw, version.Hash)
+
+	if runOpts.rootMount != "" {
+		glog.Infof(`Calling chroot("%s")`, runOpts.rootMount)
+		if err := syscall.Chroot(runOpts.rootMount); err != nil {
+			return fmt.Errorf("Unable to chroot to %s: %s", runOpts.rootMount, err)
+		}
+
+		glog.V(2).Infof("Moving to / inside the chroot")
+		if err := os.Chdir("/"); err != nil {
+			return fmt.Errorf("Unable to change directory to /: %s", err)
+		}
+	}
+
+	uri, err := url.Parse(runOpts.healthCheckURL)
+	if err != nil {
+		return fmt.Errorf("failed to parse health-check-url: %v", err)
+	}
+	if !uri.IsAbs() {
+		return fmt.Errorf("invalid URI %q (no scheme)", uri)
+	}
+
+	httpCheck, err := checkers.NewHTTP(&checkers.HTTPConfig{
+		URL: uri,
+		Client: &http.Client{Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create httpCheck: %v", err)
+	}
+
+	errCh := make(chan error)
+	tracker := &healthTracker{
+		ErrCh:            errCh,
+		SuccessThreshold: 2,
+		FailureThreshold: 10,
+		OnFailure:        func() error { return exec.Command("systemctl", "stop", runOpts.gcpRoutesService).Run() },
+		OnSuccess:        func() error { return exec.Command("systemctl", "start", runOpts.gcpRoutesService).Run() },
+	}
+
+	h := health.New()
+	h.AddChecks([]*health.Config{{
+		Name:       "dependency-check",
+		Checker:    httpCheck,
+		Interval:   time.Duration(5) * time.Second,
+		Fatal:      true,
+		OnComplete: tracker.OnComplete,
+	}})
+
+	if err := h.Start(); err != nil {
+		return fmt.Errorf("failed to start heath checker: %v", err)
+	}
+
+	for {
+		select {
+		case err := <-errCh:
+			if err != nil {
+				return fmt.Errorf("error running health checker: %v", err)
+			}
+		}
+	}
+}
+
+type trackerState int
+
+const (
+	unknownTrackerState trackerState = iota
```

Comment:
this var is unused so it's going to fail verify unless removed.

Reply:
https://github.com/openshift/machine-config-operator/pull/1031/commits/d83c4e499a093a1cf85fca2f0c314feba515a619

Reply:
:+1: